### ISSUE TITLE
[PW-4572] Issue with notifications skipped in shouldSkipProcessingNotification()

### ISSUE THREAD

**Issue by ebaschiera** [Adyen/adyen-magento2#1014]
**Describe the bug**
Hi,
at line https://github.com/Adyen/adyen-magento2/blob/7.0.0/Model/Cron.php#L419 you should not use the "`->i`" to calculate how many minutes difference there is between the two DateTime objects. Because the "i" attribute doesn't take into account the hours or even days in DateInterval.

Let's see that issue with an example. Let's say that for some reason the `adyen_payment_process_notification` cron job  doesn't run for one day. We will have an unprocessed notification from a day before.

That code will run like this:
```
$offerClosedMinDate = new \DateTime('-10 minutes'); // 2021-04-16 16:40:00 (Timezone: CEST)
$createdAt = \DateTime::createFromFormat('Y-m-d H:i:s', $notification['created_at']); // 2021-04-15 16:45:00
$minutesUntilProcessing = $createdAt->diff($offerClosedMinDate)->i; //5
```

So the notification will be skipped even if it's from a day before.

I suggest to convert those DateTime objects into seconds and make an arithmetic substraction, like this:
`$minutesUntilProcessing = ($createdAt->format('U') - $offerClosedMinDate->format('U')) / 60; //-1445`

We had several notifications stuck because the cron job was configured to run every 5 minutes (for system load mitigation purpose) and unfortunately that age check was failing consistently.

Thanks!

**Comment by acampos1916** [comment]
Hello @ebaschiera thanks for opening this issue. In production mode the [recommended cron frequency is 1min](https://docs.adyen.com/plugins/magento-2/set-up-the-plugin-in-magento#step-2-run-cron) but makes sense that this shouldn't prevent the old notifications from being processed. Opening an internal ticket to test your solution.

**Comment by zakisaad** [comment]
Dropping in as we also experience issues with this on a cron-heavy site, and have contemplated adding in code similar to the suggested by OP.

Will deal with it for now, what are the chances a fix for this will drop in the 7.0.1 release? We are holding off moving from 6.x.x until the first 7.x.x point release, would be ideal if it also included a fix for this!

TIA

**Comment by acampos1916** [comment]
Hi @zakisaad, working to include this in 7.0.1 ASAP 👍
